src/WpfMath/Atoms/RowAtom.cs

```diff
@@ -46,24 +46,26 @@ public RowAtom(SourceSpan source, Atom baseAtom)
         public RowAtom(SourceSpan source)
             : base(source)
         {
-            this.Elements = new List<Atom>().AsReadOnly();
+            this.Children = new List<Atom>().AsReadOnly();
         }
 
         private RowAtom(SourceSpan source, DummyAtom previousAtom, ReadOnlyCollection<Atom> elements)
             : base(source)
         {
             this.PreviousAtom = previousAtom;
-            this.Elements = elements;
+            this.Children = elements;
         }
 
         internal RowAtom(SourceSpan source, IEnumerable<Atom> elements)
-            : base(source) =>
-            this.Elements = elements.Where(x => x != null).ToList().AsReadOnly();
+            : base(source)
+        {
+            this.Children = elements.Where(x => x != null).ToList().AsReadOnly(); ;
+        }
             // TODO[F]: Fix this with C# 8 migration: there shouldn't be nullable atoms in this collection
 
         public DummyAtom PreviousAtom { get; }
 
-        public ReadOnlyCollection<Atom> Elements { get; }
+        public ReadOnlyCollection<Atom> Elements => (ReadOnlyCollection<Atom>)this.Children;
```

Comment:
Since you are using a `protected set`, hypothetically, the likelihood of a subsequent exception is even greater. So I would think about using a certain type

Reply:
Or am I wrong here, sorry

Reply:
Yes, this [example](https://dotnetfiddle.net/ANWesf) throws `System.InvalidCastException`

Reply:
Sorry, my fault

Reply:
>this example

Sorry, something I began to complicate simple things yesterday

> Sorry, my fault

No problems. Live and learn!  :)